diff -up lightdm-gtk-greeter-1.5.2/data/lightdm-gtk-greeter.conf.fedora lightdm-gtk-greeter-1.5.2/data/lightdm-gtk-greeter.conf
--- lightdm-gtk-greeter-1.5.2/data/lightdm-gtk-greeter.conf.fedora	2013-05-13 17:31:32.000000000 -0500
+++ lightdm-gtk-greeter-1.5.2/data/lightdm-gtk-greeter.conf	2013-05-14 07:40:33.300698120 -0500
@@ -14,7 +14,7 @@
 # keyboard = command to launch on-screen keyboard
 #
 [greeter]
-#background=
+background=/usr/share/backgrounds/day.jpg
 #theme-name=
 #icon-theme-name=
 #font-name=
