diff -up lightdm-gtk-greeter-1.1.5/data/lightdm-gtk-greeter.conf.fedora lightdm-gtk-greeter-1.1.5/data/lightdm-gtk-greeter.conf
--- lightdm-gtk-greeter-1.1.5/data/lightdm-gtk-greeter.conf.fedora	2012-04-03 15:35:21.000000000 -0500
+++ lightdm-gtk-greeter-1.1.5/data/lightdm-gtk-greeter.conf	2012-04-25 13:09:47.600046370 -0500
@@ -9,7 +9,7 @@
 # xft-rgba = Type of subpixel antialiasing (none, rgb, bgr, vrgb or vbgr)
 #
 [greeter]
-#background=
+background=/usr/share/backgrounds/default.png
 #show-language-selector=false
 #theme-name=
 #font-name=
