diff -up lightdm-gtk-greeter-1.8.5/src/lightdm-gtk-greeter.c.lp#1024482 lightdm-gtk-greeter-1.8.5/src/lightdm-gtk-greeter.c
--- lightdm-gtk-greeter-1.8.5/src/lightdm-gtk-greeter.c.lp#1024482	2014-10-06 13:52:51.869311249 -0500
+++ lightdm-gtk-greeter-1.8.5/src/lightdm-gtk-greeter.c	2014-10-06 13:54:15.032163963 -0500
@@ -2267,6 +2267,9 @@ main (int argc, char **argv)
     /* Disable global menus */
     g_unsetenv ("UBUNTU_MENUPROXY");
 
+    /* LP: #1024482 */
+    g_setenv ("GDK_CORE_DEVICE_EVENTS", "1", TRUE);
+
     /* Initialize i18n */
     setlocale (LC_ALL, "");
     bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
