diff -up lightdm-gtk-greeter-1.8.1/src/lightdm-gtk-greeter.c.validate_session lightdm-gtk-greeter-1.8.1/src/lightdm-gtk-greeter.c
--- lightdm-gtk-greeter-1.8.1/src/lightdm-gtk-greeter.c.validate_session	2014-02-20 06:50:56.371732796 -0600
+++ lightdm-gtk-greeter-1.8.1/src/lightdm-gtk-greeter.c	2014-02-20 06:53:41.145973044 -0600
@@ -528,6 +528,7 @@ set_session (const gchar *session)
     const gchar *default_session;
     gchar *last_session;
     GList *menu_items, *menu_iter;
+    GList *items, *item;
 #if GTK_CHECK_VERSION (3, 0, 0)
     GtkIconTheme *icon_theme = gtk_icon_theme_get_default();
 #endif
@@ -569,9 +570,26 @@ set_session (const gchar *session)
     last_session = g_key_file_get_value (state, "greeter", "last-session", NULL);
     if (last_session && g_strcmp0 (session, last_session) != 0)
     {
-        set_session (last_session);
-        g_free (last_session);
-        return;
+          /* Go thru all sessions and compare them to our last_session otherwise we can get a segfault
+           * if last_session is set to a non-existing or removed session
+           */
+           items = lightdm_get_sessions ();
+           for (item = items; item; item = item->next)
+           {
+                LightDMSession *session = item->data;
+                gchar *s;
+                gboolean matched;
+                s = lightdm_session_get_key (session);
+                matched = strcmp (s, last_session) == 0;
+                s = NULL;
+                g_free(s);
+                if (matched)
+                {
+                    set_session (last_session);
+                    g_free (last_session);
+                    return;
+                }
+           }
     }
     g_free (last_session);
     
